app/src/main/java/com/example/tasky/agenda/data/remote/util/Util.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.tasky.agenda.data.remote.util
+
+import android.annotation.SuppressLint
+import com.example.tasky.agenda.data.remote.model.AgendaResponse
+import com.example.tasky.agenda.presentation.model.Agenda
+import com.example.tasky.core.data.Resource
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+fun Resource<AgendaResponse>.mapToAgenda(): Resource<List<Agenda>> {
+    if (this is Resource.Error) {
+        return Resource.Error(this.errorMessage)
+    } else {
```

Comment:
Personally for this, I prefer using a "when" expression since it already has the smart casting

But it's fine